src/txn/txn_recover.c

```diff
@@ -822,12 +846,164 @@ __recovery_file_scan_prefix(WT_RECOVERY *r, const char *prefix, const char *igno
         if (ignore_suffix != NULL && WT_SUFFIX_MATCH(uri, ignore_suffix))
             continue;
         WT_RET(c->get_value(c, &config));
-        WT_RET(__recovery_setup_file(r, uri, config));
+        WT_RET(__recovery_func(r, uri, config));
     }
     WT_RET_NOTFOUND_OK(ret);
     return (0);
 }
 
+/*
+ * __recovery_file_scan_prefix --
+ *     Scan the files matching the prefix referenced from the metadata and gather information about
+ *     them for recovery.
+ */
+static int
+__recovery_file_scan_prefix(WT_RECOVERY *r, const char *prefix, const char *ignore_suffix)
+{
+    return (__recovery_metadata_iterate_func(r, prefix, ignore_suffix, __recovery_setup_file));
+}
+
+/*
+ * __metadata_check_consistency --
+ *     For each table metadata entry, check that the associated colgroup and file metadata entries
+ *     exist. If they do not exist, mark them for removal.
+ */
+static int
+__metadata_check_consistency(WT_RECOVERY *r, const char *uri, const char *config)
+{
+    WT_CONFIG cparser;
+    WT_CONFIG_ITEM ckey, cval, config_item;
+    WT_CURSOR *metac;
+    WT_DECL_RET;
+    size_t len;
+    char *cgname, *filename, *tieredname;
+    const char *format_name;
+    bool has_colgroup, has_file;
+
+    has_file = has_colgroup = true;
+    cgname = filename = tieredname = NULL;
+    format_name = uri;
+    WT_PREFIX_SKIP_REQUIRED(r->session, format_name, "table:");
+
+    WT_ERR(__wt_metadata_cursor_open(r->session, NULL, &metac));
+    /*
+     * FIXME-WT-XXXX: Add capability for cleaning complex and tiered tables. For now, only simple
```

Comment:
Do we need to handle complex tables? I thought that in complex cases customers are responsible for creating colgroups, so having only a `table:` entry in the metadata after a crash is OK.

Do we need this to handle a crash occurring during colgroup creation, where the file: entry has already been inserted into the metadata but the colgroup entry has not?